05-onchain.md

```diff
@@ -2,255 +2,294 @@
+      - SHOULD resolve the HTLC-timeout transaction by spending it to a
+      convenient address.
+        - Note: if the output is spent (as recommended), the output is
+        *resolved* by the spending transaction, otherwise it is considered
+        *resolved* by the commitment transaction itself.
+      - MUST wait until the `OP_CHECKSEQUENCEVERIFY` delay has passed (as
+      specified by the remote node's `open_channel` `to_self_delay` field)
+      before spending that HTLC-timeout output.
+  - for any committed HTLC that does NOT have an output in this commitment
+  transaction:
+    - once the commitment transaction has reached reasonable depth:
+      - MUST fail the corresponding incoming HTLC (if any).
+    - if no *valid* commitment transaction contains an output corresponding to
+    the HTLC.
+      - MAY fail the corresponding incoming HTLC sooner.
 
 ### Rationale
 
-The payment preimage either serves to prove payment (when the offering node originated the payment) or
-to redeem the corresponding incoming HTLC from another peer (when the offering node is forwarding the payment). Once a node has extracted the payment, it no longer cares about the fate of the HTLC-spending transaction itself.
+The payment preimage either serves to prove payment (when the offering node
+originated the payment) or to redeem the corresponding incoming HTLC from
+another peer (when the offering node is forwarding the payment). Once a node has
+extracted the payment, it no longer cares about the fate of the HTLC-spending
+transaction itself.
 
-In cases where both resolutions are possible (e.g., when a node receives payment success after timeout), either interpretation is
-acceptable; it is the responsibility of the recipient to spend it
-before this occurs.
+In cases where both resolutions are possible (e.g., when a node receives payment
+success after timeout), either interpretation is acceptable; it is the
+responsibility of the recipient to spend it before this occurs.
 
-We need to use our HTLC-timeout transaction to time out the HTLC
-to prevent them fulfilling it and claiming the funds, and before we can back-fail any corresponding incoming HTLC, using
-`update_fail_htlc` (presumably with reason
+We need to use local HTLC-timeout transaction to time out the HTLC to prevent them
+fulfilling it and claiming the funds, and before we can back-fail any
+corresponding incoming HTLC, using `update_fail_htlc` (presumably with reason
 `permanent_channel_failure`) as detailed in
-[BOLT 02](https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#forwarding-htlcs).  If the incoming HTLC is on-chain too,
-we simply wait for it to timeout: there's no way to signal early failure.
+[BOLT 02](https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#forwarding-htlcs).
+If the incoming HTLC is on-chain too, we simply wait for it to timeout: there's
```

Comment:
we->"the local node"?

Reply:
Or maybe this is a case of we, the document writers, in which case it should be removed.